src/Thruway/Registration.php

```diff
@@ -127,14 +139,22 @@ public function __construct(Session $session, $procedureName)
     public static function createRegistrationFromRegisterMessage(Session $session, RegisterMessage $msg)
     {
         $registration = new Registration($session, $msg->getProcedureName());
-        $options      = $msg->getOptions();
+        $options = $msg->getOptions();
 
         if (isset($options->disclose_caller) && $options->disclose_caller === true) {
             $registration->setDiscloseCaller(true);
         }
 
-        if (isset($options->thruway_multiregister) && $options->thruway_multiregister === true) {
-            $registration->setAllowMultipleRegistrations(true);
+        if (isset($options->invoke)) {
+            $registration->setInvokeType($options->invoke);
```

Comment:
As a matter of fact, now that I am re-reading the code, I believe you are correct. `setAllowMultipleRegistrations` is not really needed here.

However, I think `setInvokeType` needs a liitle bit of a change, just for the sake of readability. I will do the modifications and commit them when I get the time.